dev-docs/source/MaskWorkspace.rst

```diff
@@ -0,0 +1,101 @@
+(*An example of such a usecase would improve this article. Otherwise MaskWorkspace should actively enforce a 1:1 policy.*)
+
+This mapping seems odd at first, but serves a pragmantic purpose when interfacing with the rest of Workbench.
+*Opinion:* This implementation conflates frontend/backend representation of the data by modeling its format after what a user might expect particle data to be in.
+E.g. Each Detector has a Spectrum with data recorded over some unit, but for a SpecialWorkspace2D this value would be constant.
+
+
+The **X Data/Y Data** of the MaskWorkspace should be the **only source of truth** when determining if a Detector ID is masked.
+
+However, as a :py:obj:`mantid.dataobjects.SpecialWorkspace2D`, it also has a SpectrumInfo->DectectorInfo.
+
+This DetectorInfo also has the capability to track whether or not a given Detector ID is masked.😐
+
+The SpectrumInfo *also* can track if a Detector ID is masked but becomes unreliable for more complex mappings.
+
+**Don't use either of the above.^**
+
+These objects serve as metadata about the Workspace.
+On any other Workspace, these determine if specific Detector IDs or Spectrum are masked.
+In the case of a MaskWorkspace, relying on these instead of the main data creates indirection and uncertainty.
+*Opinion:* The masking aspect of DetectorInfo and SpectrumInfo probably shouldnt have been included on MaskWorkspace to begin with.
+
+
+**NOTE:** Typically, throughout the codebase if a Detector ID *does not exist* in a MaskWorkspace, it is considered *Masked*.
+However, it is safer to fully define and validate that your MaskWorkspace has a value assigned for each Detector ID of your target workspace.
+*Opinion:* There may be an opportunity for performance improvement if instead a mask only held Detector IDs it masked, e.g. the opposite of what is currently assumed.
+Or had an option to invert what its values meant.
+
+
+Implementation Notes
+--------------------
+
+**This document was not written by the author of MaskWorkspace.  These notes may border on speculation.**
+
+MaskWorkspace/SpecialWorkspace2D contains a map member variable `detID_to_WI`.
+This member serves in part as a cache for the map that would normally be generated as part of `MatrixWorkspace::getIndicesFromDetectorIDs`.
+This is used to track which Detector IDs are associated with which Spectrum.
+I do not reccommend doing so, but this mapping supports assiging multiple Detector IDs to the same Spectrum.
+This is a byproduct of being a :py:obj:`mantid.dataobjects.SpecialWorkspace2D`, and is applicable in other child classes such as :py:obj:`mantid.dataobjects.GroupingWorkspace`,
+If this sounds useful, maybe MaskWorkspace could be expanded to support such behavior with additional context beyond just this arbitrary map.
+But currently, it is usually expected to have a 1:1 mapping of Detector ID to Spectrum
```

Comment:
Correction: the map is onto, and frequently is one-to-one. The one-to-one is not guaranteed. Some detids are not associated with the workspace (e.g. monitors), but are part of the instrument.